timesketch/api/v1/resources.py

```diff
@@ -1850,6 +1854,100 @@ def get(self, sketch_id):
         }
         return jsonify(schema)
 
+class EventTaggingResource(ResourceMixin, Resource):
+    """Resource to fetch and set tags to an event."""
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+
+        Returns:
+            An annotation in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            msg = 'No sketch found with this ID.'
+            abort(HTTP_STATUS_CODE_NOT_FOUND, msg)
+
+        if not sketch.has_permission(current_user, 'write'):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, (
+                    'User does not have sufficient access rights to '
+                    'modify a sketch.'))
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        tag_dict = {
+            'event_count': 0,
+            'tag_count': 0,
+        }
+        datastore = self.datastore
+
+        try:
+            tags_to_add = json.loads(form.get('tag_string', ''))
+        except json.JSONDecodeError as e:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Unable to read the tags, with error: {0!s}'.format(e))
+        if not isinstance(tags_to_add, list):
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST, 'Tags need to be a list')
+
+        if not all([isinstance(x, str) for x in tags_to_add]):
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Tags need to be a list of strings')
+
+        events = form.get('events', [])
+        for _event in events:
```

Comment:
We should move this to the elastic datastore instead, seeing we are using this from more than one place. Let's do that later though.